### PR TITLE
feat(jenkins-plugins): Add support for custom registries

### PR DESCRIPTION
**Pull request by Nicolas-M**
## Changes

* Update the `jenkins-plugins` datasource to use custom registries
* Add README describing how a `jenkins-plugins` datasource works

## Context

* Provide more flexibility for pulling updates for Jenkins plugins, especially in cases where the update site is mirrored elsewhere. 

## Documentation (please check one with an [x])

- [X] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [X] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: lib/modules/datasource/jenkins-plugins/readme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The `jenkins-plugins` datasource supports lookups from the [Jenkins Update Center](https://updates.jenkins.io/) or custom registries.
+
+Custom registries defined via `registryUrl` are expected to follow the [update site layout](https://github.com/jenkins-infra/update-center2/blob/master/site/LAYOUT.md).
```
Comment (HonkingGoose) [inline lib/modules/datasource/jenkins-plugins/readme.md:3]:
```suggestion
When you define a custom registry with the `registryUrl` config option, you must follow [Jenkins' update site layout](https://github.com/jenkins-infra/update-center2/blob/master/site/LAYOUT.md).
```

We should use the word `must` to indicate things that you must do, or Renovate must have to work.

Also, we should say _who_ does _what_. I think in this example the user would set a custom registry, and then must follow some rules from Jenkin's for Renovate to work.

I've also updated the link name so it's clear where you'll end up.

File: lib/modules/datasource/jenkins-plugins/readme.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+Renovate expects `update-center.actual.json` and `plugin-versions.json` to be available under the `current` path (e.g., `https://custom.registry.renovatebot.com/current`).
```
Comment (HonkingGoose) [inline lib/modules/datasource/jenkins-plugins/readme.md:5]:
```suggestion
The `update-center.actual.json` and `plugin-versions.json` files must be in the `current` path, so Renovate can find them.
For example, the files could be put in the `https://custom.registry.renovatebot.com/current` directory.
```

File: lib/modules/manager/jenkins/readme.md [hunk 3]
```diff
@@ -1,4 +1,5 @@
 The Jenkins manager supports a custom text or YAML format of the plugin list as described [here](https://github.com/jenkinsci/plugin-installation-manager-tool#plugin-input-format).
-Only versions from the main [Jenkins update center](https://updates.jenkins.io/) are supported.
+
+Versions from the main [Jenkins update center](https://updates.jenkins.io/) are used by default.
```
Comment (HonkingGoose) [inline lib/modules/manager/jenkins/readme.md:3]:
```suggestion
By default, Renovate gets the versions from the main [Jenkins update center](https://updates.jenkins.io/).
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by viceice — COMMENTED** [review]
this isn't enough, you need to override custom registry support flag from datasource base class

**Review by viceice — APPROVED** [review]

**Review by HonkingGoose — APPROVED** [review]

**Pull request merged**